src/controllers/queryRunner.ts

```diff
@@ -736,6 +717,148 @@ export default class QueryRunner {
+                const result = data;
+                let count = 0;
+                for (let row of result) {
+                    let rowNumber = count + range.fromRow;
+                    if (rowIdToSelectionMap.has(rowNumber)) {
+                        let rowSelection = rowIdToSelectionMap.get(rowNumber);
+                        rowSelection.push(range);
+                    } else {
+                        rowIdToSelectionMap.set(rowNumber, [range]);
+                    }
+                    rowIdToRowMap.set(rowNumber, row);
+                    count += 1;
+                }
+            };
+        });
+        let p = tasks[0]();
+        for (let i = 1; i < tasks.length; i++) {
+            p = p.then(tasks[i]);
+        }
+        await p;
+
+        // Go through all rows and get selections for them
+        let allRowIds = rowIdToRowMap.keys();
+        const endColumns = this.getSelectionEndColumns(
+            rowIdToRowMap,
+            rowIdToSelectionMap,
+        );
+        const firstColumn = endColumns[0];
+        const lastColumn = endColumns[1];
+        for (let rowId of allRowIds) {
+            let row = rowIdToRowMap.get(rowId);
+            const rowSelections = rowIdToSelectionMap.get(rowId);
+
+            // sort selections by column to go from left to right
+            rowSelections.sort((a, b) => {
+                return a.fromCell < b.fromCell
+                    ? -1
+                    : a.fromCell > b.fromCell
+                      ? 1
+                      : 0;
+            });
+
+            for (let i = 0; i < rowSelections.length; i++) {
+                let rowSelection = rowSelections[i];
+
+                // Add tabs starting from the first column of the selection
+                for (let j = firstColumn; j < rowSelection.fromCell; j++) {
+                    copyString += "\t";
+                }
+                let cellObjects = row.slice(
+                    rowSelection.fromCell,
+                    rowSelection.toCell + 1,
+                );
+
+                // Remove newlines if requested
+                let cells = this.shouldRemoveNewLines()
+                    ? cellObjects.map((x) =>
+                          this.removeNewLines(x.displayValue),
+                      )
+                    : cellObjects.map((x) => x.displayValue);
+                copyString += cells.join("\t");
+
+                // Add tabs until the end column of the selection
+                for (let k = rowSelection.toCell; k < lastColumn; k++) {
+                    copyString += "\t";
+                }
+            }
+            copyString += os.EOL;
+        }
+
+        // Remove the last extra new line
+        if (copyString.length > 1) {
+            copyString = copyString.substring(
+                0,
+                copyString.length - os.EOL.length,
+            );
+        }
+
+        await this.writeStringToClipboard(copyString);
+    }
+
+    public addHeaders(
```

Comment:
Can you add doc to this and other public methods? You might also want to use a better name for this method as it also returns  the copy str as the return value but `addHeaders` doesn't indicate that.